<html>
<head>
<link href="rainbow-dev-custom/themes/github.css" rel="stylesheet" type="text/css">

</head>
<body>
  <!-- you can create a custom build with specific languages
this should go before the closing </body> -->
<script src="rainbow-dev-custom/rainbow.min.js"></script>
<script src="rainbow-dev-custom/language/generic.js"></script>
<script src="assets/js/language/shell.js"></script>
<script src="assets/js/language/python.js"></script>
<script type="text/javascript" src="https://www.gstatic.com/charts/loader.js"></script>

<div id="title">
  <h2>From Raw reads to OTUs</h2>
  <p>Paired-end read joining, quality filtering and OTU picking of Illumina MiSeq data in QIIME</p>
  <p>2015 DNA samples amplied with EMP 16S from Calvert Island (BIOMES), Juneau (NADP) and RNA from Calvert Island (BIOMES-R)</p>
</div>

<div id="chart">

<h2>Sankey diagram (AKA 'ribbon plot') of bioinformatics pipeline</h2>
<p>Plot showing the journey from raw Illumina MiSeq reads to annotated OTUs. See below for information about the bioinformatics pipeline and scripts used.</p>  
<div id="sankey_multiple" style="width: 1000px; height: 600px;"></div>

<script type="text/javascript">
  google.charts.load("current", {packages:["sankey"]});
  google.charts.setOnLoadCallback(drawChart);
   function drawChart() {
    var data = new google.visualization.DataTable();
    data.addColumn('string', 'From');
    data.addColumn('string', 'To');
    data.addColumn('number', 'Sequences');
    data.addRows([
       [ 'r_BIOMES', 'j_BIOMES', 49376 ],
       [ 'r_NADP', 'j_NADP', 32918 ],
       [ 'r_BIOMES-R', 'j_BIOMES-R', 32853 ],
       [ 'r_BIOMES', 'Unjoined', 2933 ],
       [ 'r_NADP', 'Unjoined', 1684 ],
       [ 'r_BIOMES-R', 'Unjoined', 1857 ],
       [ 'Unjoined', 'o_BIOMES', 1 ],
       [ 'Unjoined', 'o_NADP', 1 ],
       [ 'Unjoined', 'o_BIOMES-R', 1 ],
       [ 'j_BIOMES', 'f_BIOMES', 38474 ],
       [ 'j_NADP', 'f_NADP', 26015 ],
       [ 'j_BIOMES-R', 'f_BIOMES-R', 26194 ],
       [ 'j_BIOMES', 'Filtered', 10902 ],
       [ 'j_NADP', 'Filtered', 6903 ],
       [ 'j_BIOMES-R', 'Filtered', 6658 ],
       [ 'Filtered', 'o_BIOMES', 1 ],
       [ 'Filtered', 'o_NADP', 1 ],
       [ 'Filtered', 'o_BIOMES-R', 1 ],
       [ 'f_BIOMES', 'o_BIOMES', 27818 ],
       [ 'f_NADP', 'o_NADP', 21742 ],
       [ 'f_BIOMES-R', 'o_BIOMES-R', 23964 ],
       [ 'f_BIOMES', 'Unclassified', 10656 ],
       [ 'f_NADP', 'Unclassified', 4272 ],
       [ 'f_BIOMES-R', 'Unclassified', 2229 ]
    ]);

    // Set chart options
    var options = {
      width: 1000,
      height: 500
    };

    // Instantiate and draw our chart, passing in some options.
    var chart = new google.visualization.Sankey(document.getElementById('sankey_multiple'));
    chart.draw(data, options);
   }
</script>
</div>


<div id="main">
  <h2>Bioinformatics pipeline</h2>
  <p> 1. Using Flash to join paired end reads:</p>
  <pre><code data-language="shell">
FLASH-1.2.11/flash 
-M 100 #min overlap
-d /home/david/David_output 
-o "out" 
/home/david/David/BIOMES1001_S1_L001_R1_001.fastq.gz 
/home/david/David/BIOMES1001_S1_L001_R2_001.fastq.gz
  </pre></code>

  <p> 2. USEARCH Quality Filtering:</p>
  <a href="http://www.earthmicrobiome.org/emp-standard-protocols/16s-taxonomic-assignments/">EMP Guidelines</a>
  <ol>
    <li>Identify the first quality score below Q3 and truncate the read just prior to that position.</li>
    <li>Determine if the truncated sequence is at least 75% of the length of the input sequence: if yes, retain the truncated read; if no, discard the read.</li> 
    <li>Determine if the truncated sequence has any N (i.e., ambiguous base call) characters: if yes, discard the read; if no, retain the read.</li> 
</ol>

  <pre><code data-language="shell">
usearch8 
-fastq_filter David_output/joined_fastq/BIOMES1001.extendedFrags.fastq 
-fastq_trunclen 220 #truncate to 220bp *should have been fewer
-fastq_truncqual 3 #Quality score greater than 3
-fastq_maxns 1 #N less than 1
-fastaout David_output/joined_filtered_fasta/BIOMES1001.fasta
  </pre></code>

    <p> 3. Label and join FASTQ files for QIIME pipeline:</p>
  <pre><code data-language="shell">
add_qiime_labels.py 
--fasta_dir ~/David_output/joined_filtered_fasta
--mapping_fp DLB_2015_sample_map.txt 
--filename_column InputFileName 
--output_dir ~/David_output/joined_filtered_fasta_combined
  </pre></code>

     <p> 4. OTU open reference picking:</p>
  <pre><code data-language="shell">
pick_open_reference_otus.py 
--output_dir David_output/otus/ 
--input_fps David_output/joined_filtered_fasta_combined/combined_seqs.fna 
--parameter_fp uc_fast_params.txt
  </pre></code>

      <p> 5. Diversity Analysis:</p>
  <pre><code data-language="shell">
core_diversity_analyses.py 
-o David_output/cdout/ 
--recover_from_failure 
-c "Site,Plot" 
-i David_output/otus/otu_table_mc2_w_tax_no_pynast_failures.biom 
-m DLB_2015_sample_map.txt 
-t David_output/otus/rep_set.tre 
-e 1000 # 1000 OTU minimum
  </pre></code>



</body>
</html>
