<!DOCTYPE HTML>

<html>
	<head>
		<title>Bioinformatics Ribbon Diagram</title>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1" />
		<!--[if lte IE 8]><script src="assets/js/ie/html5shiv.js"></script><![endif]-->
		<link rel="stylesheet" href="assets/css/main.css" />
		<link href="assets/css/github.css" rel="stylesheet" type="text/css">

		<!--[if lte IE 8]><link rel="stylesheet" href="assets/css/ie8.css" /><![endif]-->
		<!--[if lte IE 9]><link rel="stylesheet" href="assets/css/ie9.css" /><![endif]-->
		<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.0/jquery.min.js"></script>

	
	</head>

	<body>
  <!-- you can create a custom build with specific languages
this should go before the closing </body> -->
<script src="assets/js/rainbow.min.js"></script>
<script src="assets/js/generic.js"></script>
<script src="assets/js/shell.js"></script>
<script src="assets/js/python.js"></script>
<script type="text/javascript" src="https://www.gstatic.com/charts/loader.js"></script>

		<!-- Header -->
			<header id="header">
			</header>


<!-- Main -->
	<section id="main" class="wrapper">
	<div id ="pub_title">
	<i class="icon fa-bar-chart"></i>
	<h2>Bioinformatics Workflow: From Raw reads to OTUs</h2>
	</div>
	</section>


<!-- Main -->
	<section id="main" class="wrapper">
	<div class="container">
<!-- begin publications list -->

<div id="title">
  <p>Paired-end read joining, quality filtering and OTU picking of Illumina MiSeq data in QIIME. 2015 DNA samples amplied with EMP 16S from Calvert Island (BIOMES), Juneau (NADP) and RNA from Calvert Island (BIOMES-R).</p>
</div>

<div id="chart">

<h2>Sankey diagram (AKA 'ribbon plot') of bioinformatics pipeline</h2>
<p>Plot showing the journey from raw Illumina MiSeq reads to annotated OTUs. See below for information about the bioinformatics pipeline and scripts used.</p>  
<div id="sankey_multiple" style="width: 1000px; height: 600px;"></div>

<script type="text/javascript">
  google.charts.load("current", {packages:["sankey"]});
  google.charts.setOnLoadCallback(drawChart);
   function drawChart() {
    var data = new google.visualization.DataTable();
    data.addColumn('string', 'From');
    data.addColumn('string', 'To');
    data.addColumn('number', 'Sequences');
    data.addRows([
       [ 'Raw BIOMES reads', 'Joined BIOMES reads', 49376 ],
       [ 'Raw NADP reads', 'Joined NADP reads', 32918 ],
       [ 'Raw BIOMES RNA reads', 'Joined BIOMES RNA reads', 32853 ],
       [ 'Raw BIOMES reads', 'Unjoined', 2933 ],
       [ 'Raw NADP reads', 'Unjoined', 1684 ],
       [ 'Raw BIOMES RNA reads', 'Unjoined', 1857 ],
       [ 'Unjoined', 'BIOMES OTUs', 1 ],
       [ 'Unjoined', 'NADP OTUs', 1 ],
       [ 'Unjoined', 'BIOMES RNA OTUs', 1 ],
       [ 'Joined BIOMES reads', 'Filtered BIOMES reads', 38474 ],
       [ 'Joined NADP reads', 'Filtered NADP reads', 26015 ],
       [ 'Joined BIOMES RNA reads', 'Filtered BIOMES RNA reads', 26194 ],
       [ 'Joined BIOMES reads', 'Filtered', 10902 ],
       [ 'Joined NADP reads', 'Filtered', 6903 ],
       [ 'Joined BIOMES RNA reads', 'Filtered', 6658 ],
       [ 'Filtered', 'BIOMES OTUs', 1 ],
       [ 'Filtered', 'NADP OTUs', 1 ],
       [ 'Filtered', 'BIOMES RNA OTUs', 1 ],
       [ 'Filtered BIOMES reads', 'BIOMES OTUs', 27818 ],
       [ 'Filtered NADP reads', 'NADP OTUs', 21742 ],
       [ 'Filtered BIOMES RNA reads', 'BIOMES RNA OTUs', 23964 ],
       [ 'Filtered BIOMES reads', 'Unclassified', 10656 ],
       [ 'Filtered NADP reads', 'Unclassified', 4272 ],
       [ 'Filtered BIOMES RNA reads', 'Unclassified', 2229 ]
    ]);
    // Set chart options
    var edge_colors = [
    '#28E2F7', //1. r_BIOMES
    '#105B63', // 2. j_BIOMES (4)
    '#FFD34E', // 3. r_NADP (2)
    '#7F6927', // 4. j_NADP (5)
    '#FF6244', // 5. r_BIOMES-R
    '#853323', // 6. j biomes r
    '#261724',  // 7. unjoined
    '#28E2F7', // 8. 0_biomes - because of unjoined
    '#FFD34E',  // 9. o nadp
    '#FF6244', // 10. o biomes r
    '#1C9FAD',  // 11. f biomes
    '#C4A23C', // 12. f nadp
    '#BD4932', // 13. f biomes r
    '#261724', // 14. filtered
    '#261724', // 15. unclassifed
     ];
    var node_colors = [
    '#0C4147', //1. r_BIOMES
    '#105B63', // 2. j_BIOMES (4)
    '#524919', // 3. r_NADP (2)
    '#7F6927', // 4. j_NADP (5)
    '#4F1E15', // 5. r_BIOMES-R
    '#853323', // 6. j biomes r
    '#261724',  // 7. unjoined
    '#28E2F7', // 8. 0_biomes - because of unjoined
    '#FFD34E',  // 9. o nadp
    '#FF6244', // 10. o biomes r
    '#1C9FAD',  // 11. f biomes
    '#C4A23C', // 12. f nadp
    '#BD4932', // 13. f biomes r
    '#261724', // 14. filtered
    '#261724', // 15. unclassifed
     ];
    
    var options = {
      sankey: {
        node: {
          colors: node_colors
        },
        link: {
          colorMode: 'gradient',
          colors: edge_colors
        }
      }
    };
    // Instantiate and draw our chart, passing in some options.
    var chart = new google.visualization.Sankey(document.getElementById('sankey_multiple'));
    chart.draw(data, options);
   }
</script>
</div>


<div id="main">
  <h2>Bioinformatics pipeline</h2>
  <p> 1. Using Flash to join paired end reads:</p>
  <pre><code data-language="shell">
FLASH-1.2.11/flash 
-M 100 #min overlap
-d /home/david/David_output 
-o "out" 
/home/david/David/BIOMES1001_S1_L001_R1_001.fastq.gz 
/home/david/David/BIOMES1001_S1_L001_R2_001.fastq.gz
  </code></pre>

  <p> 2. USEARCH Quality Filtering:</p>
  <a href="http://www.earthmicrobiome.org/emp-standard-protocols/16s-taxonomic-assignments/">EMP Guidelines</a>
  <ol>
    <li>Identify the first quality score below Q3 and truncate the read just prior to that position.</li>
    <li>Determine if the truncated sequence is at least 75% of the length of the input sequence: if yes, retain the truncated read; if no, discard the read.</li> 
    <li>Determine if the truncated sequence has any N (i.e., ambiguous base call) characters: if yes, discard the read; if no, retain the read.</li> 
</ol>

  <pre><code data-language="shell">
usearch8 
-fastq_filter David_output/joined_fastq/BIOMES1001.extendedFrags.fastq 
-fastq_trunclen 220 #truncate to 220bp *should have been fewer
-fastq_truncqual 3 #Quality score greater than 3
-fastq_maxns 1 #N less than 1
-fastaout David_output/joined_filtered_fasta/BIOMES1001.fasta
  </code></pre>

    <p> 3. Label and join FASTQ files for QIIME pipeline:</p>
  <pre><code data-language="shell">
add_qiime_labels.py 
--fasta_dir ~/David_output/joined_filtered_fasta
--mapping_fp DLB_2015_sample_map.txt 
--filename_column InputFileName 
--output_dir ~/David_output/joined_filtered_fasta_combined
  </code></pre>

     <p> 4. OTU open reference picking:</p>
  <pre><code data-language="shell">
pick_open_reference_otus.py 
--output_dir David_output/otus/ 
--input_fps David_output/joined_filtered_fasta_combined/combined_seqs.fna 
--parameter_fp uc_fast_params.txt
  </code></pre>

      <p> 5. Diversity Analysis:</p>
  <pre><code data-language="shell">
core_diversity_analyses.py 
-o David_output/cdout/ 
--recover_from_failure 
-c "Site,Plot" 
-i David_output/otus/otu_table_mc2_w_tax_no_pynast_failures.biom 
-m DLB_2015_sample_map.txt 
-t David_output/otus/rep_set.tre 
-e 1000 # 1000 OTU minimum
  </code></pre>

				</div>
			</section>

		<!-- Footer -->
			<footer id="footer">
				<!-- <div class="inner">
					<ul class="icons">
						<li><a href="#" class="icon fa-facebook">
							<span class="label">Facebook</span>
						</a></li>
						<li><a href="#" class="icon fa-twitter">
							<span class="label">Twitter</span>
						</a></li>
						<li><a href="#" class="icon fa-instagram">
							<span class="label">Instagram</span>
						</a></li>
						<li><a href="#" class="icon fa-linkedin">
							<span class="label">LinkedIn</span>
						</a></li>
					</ul>
					<ul class="copyright">
						<li>&amp;copy; Untitled.</li>
						<li>Images: <a href="http://unsplash.com">Unsplash</a>.</li>
						<li>Design: <a href="http://templated.co">TEMPLATED</a>.</li>
					</ul>
				</div> -->
			</footer>

		<!-- Scripts -->
			<script src="assets/js/jquery.min.js"></script>
			<script src="assets/js/skel.min.js"></script>
			<script src="assets/js/util.js"></script>
			<!--[if lte IE 8]><script src="assets/js/ie/respond.min.js"></script><![endif]-->
			<script src="assets/js/main.js"></script>

	</body>
</html>
